testing/test_collection.py

```diff
@@ -602,43 +620,44 @@ def test_method(self):
 
 
 class Test_getinitialnodes:
-    def test_global_file(self, testdir, tmpdir) -> None:
-        x = tmpdir.ensure("x.py")
-        with tmpdir.as_cwd():
-            config = testdir.parseconfigure(x)
-        col = testdir.getnode(config, x)
+    def test_global_file(self, pytester: Pytester) -> None:
+        tmpdir = pytester.path
+        x = ensure_file(tmpdir / "x.py")
+        with tmpdir.cwd():
+            config = pytester.parseconfigure(x)
+        col = pytester.getnode(config, x)
         assert isinstance(col, pytest.Module)
         assert col.name == "x.py"
         assert col.parent is not None
         assert col.parent.parent is None
-        for col in col.listchain():
+        for col in col.listchain():  # type: ignore[assignment]
             assert col.config is config
```

Comment:
```suggestion
        for parent in col.listchain():
            assert parent.config is config
```

The issue here happens because the name `col` is reused -- when it gets assigned the first time mypy assigns it the type of the right-hand side, and here it doesn't match the `listchain()` type. By using a different name the issue is avoided.

Reply:
Okay I got it. The error was coming because of reassignment. I have made this change.